### PR TITLE
test: first happy path

### LINKED ISSUE #680: Create a user journey test from sending a mentorship request until the mentee completes a task
## Description

As a developer,
I need to test a specific user path,
so that I can make sure a specific path can be executed usefully in a deterministic way.

This test should be located in a specific folder for separation concerns, it could be called `tests/user_journey`.
Next, I'll define a very broad scenario, but simple enough for you to start.

**Scenario:** User A (mentor) and User B (mentee)
1. User B (mentee) sends a request to User A (mentor)
2. User A (mentor) accepts request
3. User A (mentor) or User B (mentee) creates a task
4. User B (mentee) completes the task

### Update [Required]
- [ ] Create 1 user journey test that follows the scenario described above
- [ ] Document the test result on Test Quality (@anitab-org/qa-team can help)

## Definition of Done
- [ ] All of the required items are completed.
- [x] Approval by 1 mentor.

## Estimation
4 hours

### PR DESCRIPTION
**Pull request by isabelcosta**
### Description

I am creating the first user journey test following the scenario described in the issue:
```
Scenario: User A (mentor) and User B (mentee) sending 
a mentorship request until the mentee completes a task

- User B (mentee) sends a request to User A (mentor)
- User A (mentor) accepts request
- User A (mentor) or User B (mentee) creates a task
- User B (mentee) completes the task
- User B (mentee) comments on the task (TODO)
- User A (mentor) comments on the task (TODO)
```

Fixes #680

I have been formating the code using `black`:
```
black tests/user_journey/test_happy_path_1.py
```

### Type of Change:

- Code
- Quality Assurance

### How Has This Been Tested?

I am running the specific Happy Path 1 test standalone.
```
python -m unittest tests/user_journey/happy_path_1.py
```

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged
- [ ] Update Postman API at /docs folder
- [ ] Update Swagger documentation and the exported file at /docs folder
- [ ] Update requirements.txt

**Code/Quality Assurance Only**

- [x] My changes generate no new warnings 
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been published in downstream modules

### CODE REVIEW

**Review by SanketDG — COMMENTED** [review]
This looks good! Simple and straight-forward!

I assume some work is still left?

**Comment by isabelcosta** [comment]
@SanketDG nice question I wanted to clarify here. Yesterday in Office hours with @rpattath we agreed that for now this is a good example for other testers/developers to follow. I left the TODO, because it can be enhanced, but I am leaving this on purpose, just to have this one as an example. What do you think?

> This looks good! Simple and straight-forward!

Thank you so much this is very nice to read about a PR :)

**Review by rpattath — DISMISSED** [review]
LGTM, thank you Isabel !

File: tests/user_journey/test_happy_path_1.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+        self.assertEqual(MentorshipRelationState.ACCEPTED, request_state)
+
+        # - User A (mentor) or User B (mentee) creates a task
+        # POST /mentorship_relation/{request_id}/task
+
+        task_request_body = json.dumps(dict(description=self.test_description))
+        task_creation_response = self.client.post(
+            f"/mentorship_relation/{request_id}/task",
+            headers=mentor_auth_header,
+            # follow_redirects=True,
```
Comment (SanketDG) [inline tests/user_journey/test_happy_path_1.py]: nit: this is not really required right?

File: tests/user_journey/test_happy_path_1.py [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+            f"/mentorship_relation/{request_id}/accept", headers=mentor_auth_header
+        )
+
+        self.assertEqual(200, accept_response.status_code)
+
+        mentee_current_relation = self.client.get(
+            f"/mentorship_relations/current", headers=mentor_auth_header
+        )
+        mentor_current_relation = self.client.get(
+            f"/mentorship_relations/current", headers=mentor_auth_header
```
Comment (SanketDG) [inline tests/user_journey/test_happy_path_1.py:115]: I assume this was intended to be `mentee_auth_header`?

File: tests/user_journey/test_happy_path_1.py [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+        self.mentee.is_email_verified = True
+
+        self.mentor.available_to_mentor = True
+        self.mentee.need_mentoring = True
+
+        db.session.add(self.mentor)
+        db.session.add(self.mentee)
+        db.session.commit()
+
+        self.test_description = "A nice task description"
```
Comment (SanketDG) [inline tests/user_journey/test_happy_path_1.py:63]:
```suggestion
        self.task_description = "A nice task description"
```

Is that okay?

**Review by SanketDG — CHANGES_REQUESTED** [review]
Looks good apart from a few nits!

File: tests/user_journey/test_happy_path_1.py (hunk 2 above)
Comment (isabelcosta) [reply tests/user_journey/test_happy_path_1.py:115]: 🙈  what an awesome catch! @SanketDG 👏 🎉

File: tests/user_journey/test_happy_path_1.py [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+        # POST /mentorship_relation/{request_id}/accept
+
+        accept_response = self.client.put(
+            f"/mentorship_relation/{request_id}/accept", headers=mentor_auth_header
+        )
+
+        self.assertEqual(200, accept_response.status_code)
+
+        mentee_current_relation = self.client.get(
+            f"/mentorship_relations/current", headers=mentor_auth_header
```
Comment (isabelcosta) [inline tests/user_journey/test_happy_path_1.py]:
```suggestion
            f"/mentorship_relations/current", headers=mentee_auth_header
```

File: tests/user_journey/test_happy_path_1.py [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+        self.assertEqual(MentorshipRelationState.ACCEPTED, request_state)
+
+        # - User A (mentor) or User B (mentee) creates a task
+        # POST /mentorship_relation/{request_id}/task
+
+        task_request_body = json.dumps(dict(description=self.test_description))
+        task_creation_response = self.client.post(
+            f"/mentorship_relation/{request_id}/task",
+            headers=mentor_auth_header,
+            # follow_redirects=True,
```
Comment (isabelcosta) [inline tests/user_journey/test_happy_path_1.py]:
```suggestion
```

**Comment by isabelcosta** [comment]
@SanketDG done with the appropriate changes! Thank you a lot for the thoughtful review 🙌

**Comment by isabelcosta** [comment]
I guess I have to fix this error 🤦‍♀️  but I am not sure how the run tests passed and this no...
image

**Review by rpattath — APPROVED** [review]
LGTM

**Review by SanketDG — APPROVED** [review]
Gah, this got slipped, looks good!

**Comment by isabelcosta** [comment]
Thank you so much @rpattath and @SanketDG for your approvals 🎉 🙌

**Pull request merged**